plugins/repository-hdfs/build.gradle

```diff
@@ -6,376 +6,184 @@
+  api 'commons-cli:commons-cli:1.2'
+  api "commons-codec:commons-codec:${versions.commonscodec}"
+  api 'commons-io:commons-io:2.8.0'
+  api 'org.apache.commons:commons-lang3:3.11'
+  api 'javax.servlet:javax.servlet-api:3.1.0'
+  api "org.slf4j:slf4j-api:${versions.slf4j}"
+  runtimeOnly "org.slf4j:slf4j-nop:${versions.slf4j}"
+  //  runtimeOnly("org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}") https://github.com/elastic/elasticsearch/issues/93714
+
+//    krb5Keytabs project(path: ':test:fixtures:krb5kdc-fixture', configuration: 'krb5KeytabsHdfsDir')
+//    krb5Config project(path: ':test:fixtures:krb5kdc-fixture', configuration: 'krb5ConfHdfsFile')
+
+
+  testImplementation(project(':test:fixtures:hdfs2-fixture')) {
+    transitive = false
+  }
+
+  javaRestTestCompileOnly(project(':test:fixtures:hdfs2-fixture')) {
+    transitive = false
+  }
+  javaRestTestImplementation project(':test:fixtures:krb5kdc-fixture')
+  javaRestTestImplementation "org.slf4j:slf4j-api:${versions.slf4j}"
+  javaRestTestRuntimeOnly "com.google.guava:guava:16.0.1"
+  javaRestTestRuntimeOnly "commons-cli:commons-cli:1.2"
+  javaRestTestRuntimeOnly "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
+
+
+  hdfsFixture2 project(path: ':test:fixtures:hdfs2-fixture', configuration: 'shadowedHdfs2')
+  hdfsFixture3 project(path: ':test:fixtures:hdfs2-fixture', configuration: 'shadow')
+  yamlRestTestCompileOnly(project(':test:fixtures:hdfs2-fixture')) {
+    transitive = false
+  }
+  yamlRestTestImplementation project(':test:fixtures:krb5kdc-fixture')
+  yamlRestTestImplementation "org.slf4j:slf4j-api:${versions.slf4j}"
+  yamlRestTestRuntimeOnly "com.google.guava:guava:16.0.1"
+  yamlRestTestRuntimeOnly "commons-cli:commons-cli:1.2"
+  yamlRestTestRuntimeOnly "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
 }
 
 restResources {
-    restApi {
-        include '_common', 'cluster', 'nodes', 'indices', 'index', 'snapshot'
-    }
-}
-
-normalization {
-    runtimeClasspath {
-        // ignore generated keytab files for the purposes of build avoidance
-        ignore '*.keytab'
-        // ignore fixture ports file which is on the classpath primarily to pacify the security manager
-        ignore 'ports'
-    }
+  restApi {
+    include '_common', 'cluster', 'nodes', 'indices', 'index', 'snapshot'
+  }
 }
 
 tasks.named("dependencyLicenses").configure {
-    mapping from: /hadoop-.*/, to: 'hadoop'
+  mapping from: /hadoop-.*/, to: 'hadoop'
 }
 
-// TODO work that into the java rest test plugin when combined with java plugin
-sourceSets {
-    javaRestTest {
-        compileClasspath = compileClasspath + main.compileClasspath
-        runtimeClasspath = runtimeClasspath + main.runtimeClasspath + files("src/main/plugin-metadata")
-    }
+tasks.withType(RestIntegTestTask).configureEach {
+  usesDefaultDistribution()
+  jvmArgs '--add-exports', 'java.security.jgss/sun.security.krb5=ALL-UNNAMED'
 }
 
 tasks.named('javaRestTest').configure {
```

Comment:
instead of disabling javaRestTest and have  javaRestTest2 and javaRestTest3 I tweaked it so javaRestTest is kind of the default (testing against hdfs3) and in addition we have another javaRestTest for tests against hdfs2.

Same logic is applied to yamlRestTests